src/core/qgsimagecache.cpp

```diff
@@ -102,7 +102,7 @@ QgsImageCache::QgsImageCache( QObject *parent )
   }
   else
   {
-    const int sysMemory = QgsApplication::systemMemorySizeMb();
+    const unsigned long sysMemory = QgsApplication::systemMemorySizeMb();
     if ( sysMemory > 0 )
     {
       if ( sysMemory >= 32000 ) // 32 gb RAM (or more) = 500mb cache size
```

Comment:
Probably doesn't belong in this PR but the units are wrong.  MB, not mb which is millibits :-)
Also, the comment and the code don't match, as 32 GB of ram is 32768 MB.

Perhaps more seriously, this is checking for >= 32 GB, but a machine with 32 GB will have physmem64 being less, presumably due to memory stolen by the BIOS for frame bufffer, or the kernel.

On NetBSD 10 amd64 with 32 GB of actual RAM and no virtualization involved.
```
$ sysctl hw.physmem; sysctl hw.physmem64
hw.physmem = -1
hw.physmem64 = 34159534080
```
which is 32577.07 MB, 190.9 MB short of 32G.  That matches the kernel's available memory from boot:
```
total memory = 32577 MB
avail memory = 31474 MB
```

So checking for "32GB" probably should be "> 30 GB".